File: examples/future/c/templates/scanner.c
int transition_{{ scanner.token }}(int state, int input)
{
    fail_if_empty(input);
    switch (state) {
## for state in scanner.transitions
    case {{ state }}:
## for transition in scanner.transitions[state]
## set start = transition[0].start
## set end = transition[0].end
## set next_state = transition[1]
## if start == end
        if (input == {{ start }})
## else
        if ({{ start }} <= input && input <= {{ end }})
## endif
            return {{ next_state }};
## endfor
        return -1;
## endfor
    default:
        return -1;
    }
}

File: examples/future/python/templates/scanner.py
class {{ scanner.token|title }}Scanner(Scanner):
    def __init__(self):
        super().__init__()
        self.token = "{{ scanner.token }}"

## for state in scanner.transitions

    def s{{ state }}(self, char):
## for transition in scanner.transitions[state]
## set start = transition[0].start
## set end = transition[0].end
## set next_state = transition[1]
## if start == end
        if char == {{ start }}:
## else
        if {{ start }} <= char <= {{ end }}:
## endif
## if next_state in scanner.accepts
            return self.change_state({{ next_state }}, checkpoint=True)
## else
            return self.change_state({{ next_state }})
## endif
## endfor
        return self.change_state(-1)

## endfor
